pkg/model/components/kubeproxy.go

```diff
@@ -68,8 +68,12 @@ func (b *KubeProxyOptionsBuilder) BuildOptions(o interface{}) error {
 		// If we're not using the AmazonVPC networking, and the KubeControllerMananger has
 		// a ClusterCIDR, use that because most networking plug ins draw pod IPs from this range.
 		if clusterSpec.Networking.AmazonVPC == nil && clusterSpec.KubeControllerManager != nil {
-			config.ClusterCIDR = clusterSpec.KubeControllerManager.ClusterCIDR
+			config.ClusterCIDR = &clusterSpec.KubeControllerManager.ClusterCIDR
 		}
+		// The user can explicitly omit --cluster-cidr from the kube-proxy configuration
+		// by setting spec.kubeProxy.clusterCIDR: ""
+	} else if *config.ClusterCIDR == "" {
+		config.ClusterCIDR = nil
```

Comment:
If something like this is necessary, it should be done just before turning into flags, in `KubeProxyBuilder`.

Reply:
Not necessary - removed